app/models/user.rb

```diff
@@ -342,14 +342,29 @@ def destroy
     super
   end
 
-  %w[veteran_status military_information payment].each do |emis_method|
-    define_method(emis_method) do
-      emis_model = instance_variable_get(:"@#{emis_method}")
-      return emis_model if emis_model.present?
-
-      emis_model = "EMISRedis::#{emis_method.camelize}".constantize.for_user(self)
-      instance_variable_set(:"@#{emis_method}", emis_model)
-      emis_model
+  military_information_method = 'military_information'
+  define_method(military_information_method) do
+    model = instance_variable_get(:"@#{military_information_method}")
+    return model if model.present?
+
+    form_profile = FormProfile.new(form_id: nil, user: self)
+    model = form_profile.initialize_military_information
+    instance_variable_set(:"@#{military_information_method}", model)
+    model
+  end
+
+  def veteran_status
+    if Flipper.enabled?(:veteran_status_updated)
+      @veteran_status ||= VAProfile::VeteranStatus::Service.new(self)
```

Comment:
It'll make your life easier if you just pass `edipi` in here, no need for the rest of the `User` object.

Reply:
Good idea